### PR TITLE
moved common classes to base class and now extending it in respective controllers

### LINKED ISSUE #26: move all common classes to base controller and extend it in respective controllers

### PR DESCRIPTION
**Pull request by nangchetnamaw**
Fixes #26

### CODE REVIEW

File: api/controllers/base.js [hunk 1]
```diff
@@ -16,12 +16,135 @@ class Base {
     }
 
     async modify(req, res){
+        console.log(req, res, ' req, res ');
         await this.model.modify(req.params.id, req.body);
```
Comment (sonalichawla16) [inline api/controllers/base.js:20]: Please remove console logs.

File: api/controllers/base.js [hunk 2]
```diff
@@ -16,12 +16,135 @@ class Base {
...
+
+    async delete(req,res){
+        try{
+            const objToDelete = await this.model.findOne({_id: req.params.id});
+            if(objToDelete==null){
+                res.send({
+                    success: true,
+                    payload: {
+                        message: "no record found"
+                    }
```
Comment (sonalichawla16) [inline api/controllers/base.js:93]: Make the messages in payload an more meaning and with casing, as we will be showing these in alerts on UI frontend.

File: api/controllers/interview.js [hunk 3]
```diff
@@ -1,110 +1,11 @@
...
-                        message: "created interview successfully"
-                    }
-                });
-
-        }
-        catch(error){
-            console.log(error);
-        }
+        super(interviewModel);
+        console.log('Child Constructor');
```
Comment (sonalichawla16) [inline api/controllers/interview.js:7]: Remove console logs

File: api/controllers/base.js (hunk 2 above)
Comment (nangchetnamaw) [reply api/controllers/base.js:93]: okay

File: api/controllers/interview.js (hunk 3 above)
Comment (nangchetnamaw) [reply api/controllers/interview.js:7]: okay

File: api/controllers/base.js (hunk 1 above)
Comment (nangchetnamaw) [reply api/controllers/base.js:20]: okay

File: api/controllers/base.js [hunk 4]
```diff
@@ -22,6 +22,128 @@ class Base {
...
+    async create(req,res){
+        try{    
+            let objToCreate=req.body;
+            let createdObj = await this.model.save(objToCreate);
+            return res.send({
+                    success: true,
+                    payload: {
+                        body: createdObj,
+                        message: "Successfully created"
+                    }
```
Comment (sonalichawla16) [inline api/controllers/base.js:35]: meaningful messages

File: api/controllers/base.js (hunk 2 above)
Comment (sonalichawla16) [reply api/controllers/base.js:92]: Follow same pattern of meesage thorughout all APIs, No **** records found

Comment (nangchetnamaw) [reply api/controllers/base.js:126]: done

File: api/controllers/base.js (hunk 4 above)
Comment (nangchetnamaw) [reply api/controllers/base.js:33]: okay

**Comment by sonalichawla16** [comment]
Conflicts again.
I will make sure to merge it first PR.

**Comment by nangchetnamaw** [comment]
resolved all conflicts please merge it

**Comment by nangchetnamaw** [comment]
resolved all conflicts

**Pull request merged**